src/sync/sync_session.cpp

```diff
@@ -899,3 +820,95 @@ void SyncSession::did_drop_external_reference()
 
     m_state->close(lock, *this);
 }
+
+uint64_t SyncProgressNotifier::register_callback(std::function<SyncProgressNotifierCallback> notifier,
+                                                 bool is_download, bool is_streaming)
```

Comment:
Changing this argument from an enum to a boolean makes the call sites in the tests harder to decipher.

Reply:
Yeah, I think I'll revert that. I intended to translate at the public API boundary, but then I moved where the relevant API boundary was and didn't reconsider...